File: Calto-GG/Goldman Sachs/Day 2/Minimum_Consecutive_Cards_to_Pick_Up.py

# Simple intuative approach -> Just check for the keys and value of length in 
# dictionary and have an array for multiple keys and return its minimun 
class Solution:
    def minimumCardPickup(self, cards: list[int]) -> int:
        dic = {}
        lengths = []
        for i,j in enumerate(cards):
            if j in dic:
                lengths.append(abs(dic[j]-i)+1)
                dic[j]=i
            else:
                dic[j]=i
        if len(lengths)>0:
            return min(lengths)
        else:
            return -1

File: Sudesh_R/Goldman Sachs/Day7/minimum-consecutive-cards-to-pick-up.cpp

class Solution {
public:
    int minimumCardPickup(vector<int>& c) {
        vector<int>v(1000005 , -1);
        int mn = INT_MAX;
        for(int i = 0;i<c.size() ;i++)
        {
            if(v[c[i]]!=-1)
            {
                mn = min(mn , (i - v[c[i]]) +1);
                v[c[i]] = i;
            }
            else
            {
                v[c[i]] = i;
            }
        }
        if(mn==INT_MAX)
        {
            return -1;
        }

        return mn;


    }
};
